### ISSUE TITLE
local builds not written to (local) remote-storage

### ISSUE THREAD

**Issue by rqwa** [alisw/alibuild#299]
When running a local build with `aliBuild -z -d -w ../sw build AliPhysics --jobs 24  --remote-store /data02/aliceexp/aliroot/alibuild/store/::rw --disable fastjet` the following output appears:

> Write store disabled since some packages will be picked up from local checkout.
> Local packages: AliRoot AliPhysics
> DEBUG: Main package is AliPhysics@0
> ......
> DEBUG:AliPhysics:0: sending incremental file list 
> rsync: change_dir "/data02/aliceexp/aliroot/alibuild/store//TARS/slc6_x86-64/store/61/6196710596f85e564da2aa10112d2928421e4b42" failed: No such file or directory (2)
> rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1039) [sender=3.0.6]
> DEBUG:AliPhysics:0:
> DEBUG:AliPhysics:0: sent 18 bytes  received 12 bytes  60.00 bytes/sec
> DEBUG:AliPhysics:0: total size is 0  speedup is 0.00
> DEBUG:AliPhysics:0: sending incremental file list
> rsync: change_dir "/data02/aliceexp/aliroot/alibuild/store//TARS/slc6_x86-64/defaults-release" failed: No such file or directory (2)
> rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1039) [sender=3.0.6]
> DEBUG:AliPhysics:0:
> DEBUG:AliPhysics:0: sent 18 bytes  received 12 bytes  60.00 bytes/sec
> DEBUG:AliPhysics:0: total size is 0  speedup is 0.00
> DEBUG:AliPhysics:0:
> DEBUG:AliPhysics:0: Checking for packages already built.
> DEBUG:AliPhysics:0: Glob pattern used: /data02/aliceexp/marquard/software/store_test/sw/TARS/slc6_x86-64/defaults-release/defaults-release-v1-*.slc6_x86-64.tar.gz
> DEBUG:AliPhysics:0: Packages already built using this version
> DEBUG:AliPhysics:0: /data02/aliceexp/marquard/software/store_test/sw/TARS/slc6_x86-64/defaults-release/defaults-release-v1-1.slc6_x86-64.tar.gz
> Package defaults-release with hash 6196710596f85e564da2aa10112d2928421e4b42 is already found in /data02/aliceexp/marquard/software/store_test/sw/TARS/slc6_x86-64/defaults-release/defaults-release-v1-1.slc6_x86-64.tar.gz. Not building     .
> DEBUG:AliPhysics:0: Package defaults-release was correctly compiled. Moving to next one. 

This output is the same for each package.
Also only the folders `dist` `dist-direct` `dist-runtime` are created at the store folder. Those folders all contain broken links:

> ll dist*/defaults-release/defaults-release-v1-1/
> dist/defaults-release/defaults-release-v1-1/:
> total 1
> lrwxrwxrwx 1 marquard aliceexp 122 Jun  2 13:26 defaults-release-v1-1.slc6_x86-64.tar.gz -> ../../../../../TARS/slc6_x86-64/store/61/6196710596f85e564da2aa10112d2928421e4b42/defaults-release-v1-1.slc6_x86-64.tar.gz
> dist-direct/defaults-release/defaults-release-v1-1/:
> total 1
> lrwxrwxrwx 1 marquard aliceexp 122 Jun  2 13:26 defaults-release-v1-1.slc6_x86-64.tar.gz -> ../../../../../TARS/slc6_x86-64/store/61/6196710596f85e564da2aa10112d2928421e4b42/defaults-release-v1-1.slc6_x86-64.tar.gz
> dist-runtime/defaults-release/defaults-release-v1-1/:
> total 1
> lrwxrwxrwx 1 marquard aliceexp 122 Jun  2 13:26 defaults-release-v1-1.slc6_x86-64.tar.gz -> ../../../../../TARS/slc6_x86-64/store/61/6196710596f85e564da2aa10112d2928421e4b42/defaults-release-v1-1.slc6_x86-64.tar.gz 

Creating the store folder by hand also does not solve the problem.

**Comment by rqwa** [comment]
It seems that the write store is disabled for all packages if one package is a development package.
```
 734   if develPkgs:
 735     info("Write store disabled since some packages will be picked up from local checkout.")
 736     info("Local packages: %s" % " ".join(develPkgs))
 737     syncHelper.writeStore = ""
```
If I understood Giulio right, the write store should only be disabled for the development package.

**Comment by ktf** [comment]
mmm... Maybe in the end I disabled it "tout court" because it was too complicated to have a sensible setup... In principle what it should do is to disable the devel package and all it's dependencies. Maybe I was just lazy implementing the logic.

**Comment by rqwa** [comment]
7083c5dc093270e04da145f1ff3946b70cf8562b should solve the issue. But it will only work if the package was compiled  in the current run of aliBuild.
Maybe the following lines of code would result in a check if all packages are uploaded to the remote storage everytime alibuild is called with a remote/write storage.
```
 960     fileHash = readHashFile(hashFile)
 961     if fileHash != spec["hash"]:
 962       if fileHash != "0":
 963         debug("Mismatch between local area and the one which I should build. Redoing.")
 964       shutil.rmtree(dirname(hashFile), True)
 965     else:
 966       # If we get here, we know we are in sync with whatever remote store.  We
 967       # can therefore create a directory which contains all the packages which
 968       # were used to compile this one.
...
1017         except:
1018           pass
1019       syncHelper.syncToRemote(p, spec)
1020       continue
```

**Comment by ktf** [comment]
I think I fixed it in #398.
